src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -513,6 +513,9 @@ void GcsNodeManager::UpdateAliveNode(
     return;
   }
 
+  *maybe_node_info.value()->mutable_resources_available() =
```

Comment:
This information is not published via pubsub so the `node_cache_` in `NodeInfoAccessor` won't have `resources_available` set. This means people shouldn't use `resources_available` of GcsNodeInfo returned from `NodeInfoAccessor::Get()` since it won't be populated. How do we make sure people won't accidentally use it and be surprised. Should this be published via pubsub so `node_cache_` is up-to-date as well? cc @dayshah @edoakes

Reply:
We want to remove Serve's dependency on `ray.available_resources_per_node` because it's not a public API.

Reply:
Yeah sounds like we should have two separate pubsub channels -- node liveness/membership and node details